### PR TITLE
Restart task in RTM

### PR DESCRIPTION
**Pull request by Krigpl**

### CODE REVIEW

**Review by maaktweluit — APPROVED** [review]
LGTM!

Nice test cleanup

File: golem/task/requestedtaskmanager.py [hunk 1]
```diff
@@ -358,6 +358,11 @@ def get_subtask_outputs_dir(self, task_id: TaskId) -> Path:
 
         await self._shutdown_app_client(task.app_id)
 
+    async def restart_task(self, task_id: TaskId) -> None:
+        task = RequestedTask.get(RequestedTask.task_id == task_id)
+        task.status = TaskStatus.waiting
+        task.save()
```
Comment (maaktweluit) [inline golem/task/requestedtaskmanager.py:364]: should we also discard the completed results?

**Review by maaktweluit — CHANGES_REQUESTED** [review]

File: golem/task/requestedtaskmanager.py (hunk 1 above)
Comment (Krigpl) [reply golem/task/requestedtaskmanager.py:364]: There are multiple ways of task restarts we wanna support
- restart just the failed subtasks - that's the one
- restart the whole task by creating a new one from scratch
- restart particular subtasks

Comment (maaktweluit) [reply golem/task/requestedtaskmanager.py:364]: ok, should we make issues for the other 2? Do we want to support all 3 from the first release?

**Review by maaktweluit — APPROVED** [review]

**Pull request merged**